wagtail/models/revisions.py

```diff
@@ -258,6 +258,28 @@ class RevisionMixin(models.Model):
         editable=False,
     )
 
+    _revisions = GenericRelation(
+        "wagtailcore.Revision",
+        content_type_field="content_type",
+        object_id_field="object_id",
+        for_concrete_model=False,
+    )
+    """
+    A default ``GenericRelation`` for the purpose of automatically deleting
+    revisions when the object is deleted. This is not used to query the object's
+    revisions. Instead, the :meth:`revisions` property is used for that purpose.
+    As such, this default relation is considered private.
+
+    This ``GenericRelation`` does not have a
+    :attr:`~django.contrib.contenttypes.fields.GenericRelation.related_query_name`,
+    so it cannot be used for reverse-related queries from ``Revision`` back to
+    this model. If the feature is desired, subclasses can define their own
+    ``GenericRelation`` to ``Revision`` with a custom ``related_query_name``.
```

Comment:
Not sure whether we should explain this vs just defining a default `related_query_name="%(app_label)s_%(class)ss",`. Django has quite a few quirks around GFK/GenericRelations, such as: if you use the `GenericRelation` directly in a query, e.g. doing a

```py
MyModel.objects.filter(_revisions__object_str__icontains="old title")
```

You'll get duplicate results of the same `MyModel` object for each revision that matches. This is a result of the table `JOIN`, so it's not really a problem specific to `GFK/GenericRelation`s. But due to these quirks, I'm leaning towards not advising its usage by not exposing the `related_query_name`, and leaving that to developers if they so wish to use the feature (knowing the quirks come from Django rather than Wagtail).

Reply:
I think it would be better that wagtail inherit the quarks from django...